libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinGradleSubplugin.kt

```diff
@@ -85,6 +92,31 @@ interface KotlinGradleSubplugin<in KotlinCompile : AbstractCompile> {
     fun getNativeCompilerPluginArtifact(): SubpluginArtifact? = null
 }
 
+/**
+ * Gradle plugin implementing support for a Kotlin compiler plugin.
+ * In order to be discovered, it should be applied to the project as an ordinary [Plugin].
+ * Then its [isApplicable] is checked against compilations of the project, and if it returns true,
+ * then [applyToCompilation], [getPluginKotlinTasks], and other functions may be called later.
+ */
+interface KotlinCompilerPluginSupportPlugin : Plugin<Project> {
+    override fun apply(project: Project) = Unit
+
+    fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean
+
+    fun applyToCompilation(
+        kotlinCompilation: KotlinCompilation<*>,
+        javaCompile: TaskProvider<out AbstractCompile>?
+    ): Provider<List<SubpluginOption>>
+
+    fun getPluginKotlinTasks(
```

Comment:
A convention is to distinguish methods returning tasks providers (from methods returning tasks) by suffix them:
`getPluginKotlinTaskProviders` here.